From a93f60498176d86f25cadd864b0719bc786087db Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Sat, 25 Jan 2025 17:28:06 +0800
Subject: [PATCH 4/7] FEDORA: Return the expected GObject * type

This was not fixed upstream before this file was removed.

Link: https://src.fedoraproject.org/rpms/gtk2/blob/a3a398f4f95684dba7a461acf389626eb2a3b90e/f/gtk2-c89-4.patch
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 gtk/gtkuimanager.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gtk/gtkuimanager.c b/gtk/gtkuimanager.c
index d004cf1ed3..8cbeda9e85 100644
--- a/gtk/gtkuimanager.c
+++ b/gtk/gtkuimanager.c
@@ -503,7 +503,7 @@ gtk_ui_manager_buildable_construct_child (GtkBuildable *buildable,
   g_signal_connect (widget, "hierarchy-changed",
 		    G_CALLBACK (child_hierarchy_changed_cb),
 		    GTK_UI_MANAGER (buildable));
-  return g_object_ref (widget);
+  return G_OBJECT (g_object_ref (widget));
 }
 
 static void
-- 
2.48.1

